ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Run.hs

```diff
@@ -238,9 +241,12 @@ runPointSchedule schedulerConfig genesisTest tracer0 =
     fetchClientRegistry <- newFetchClientRegistry
     let chainDbView = CSClient.defaultChainDbView chainDb
     for_ (psrPeers resources) $ \PeerResources {prShared, prChainSync, prBlockFetch} -> do
-      startChainSyncConnectionThread registry tracer config chainDbView fetchClientRegistry prShared prChainSync chainSyncTimeouts_ chainSyncLoPBucketConfig stateViewTracers (psrCandidates resources)
-      BlockFetch.startKeepAliveThread registry fetchClientRegistry (srPeerId prShared)
-      startBlockFetchConnectionThread registry fetchClientRegistry (pure Continue) prShared prBlockFetch
+      forkLinkedThread registry ("Peer overview " ++ (show $ srPeerId prShared)) $
+        withRegistry $ \peerRegistry -> do
```

Comment:
Maybe add a comment explaining the purpose of this arrangement. e.g.
```
-- The peerRegistry helps ensuring that if any thread fails, then
-- the registry is closed and all threads related to the peer are
-- killed.
```

Reply:
Done in latest version